### PR TITLE
feat(ctb): Set L1 head hash in DGF rather than the FDG

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Sets the L1 head hash in the `DisputeGameFactory` rather than the `FaultDisputeGame`'s `initialize` function.

**Metadata**
closes https://github.com/ethereum-optimism/client-pod/issues/567

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9499** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9499?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9499?utm_source=stack-comment).

**Review by ajsutton — APPROVED** [review]
Winning. Thanks. :)

File: packages/contracts-bedrock/test/dispute/DisputeGameFactory.t.sol [hunk 1]
```diff
@@ -350,6 +350,10 @@ contract FakeClone {
         return hex"FF0420";
     }
 
+    function parentHash() external pure returns (bytes32) {
```
Comment (Inphi) [inline packages/contracts-bedrock/test/dispute/DisputeGameFactory.t.sol:353]: this isn't needed is it?

**Review by Inphi — APPROVED** [review]
LGTM!
A nit on an unused function.

File: packages/contracts-bedrock/test/dispute/DisputeGameFactory.t.sol (hunk 1 above)
Comment (clabby) [reply packages/contracts-bedrock/test/dispute/DisputeGameFactory.t.sol:353]: Not necessarily - just put it there for completion. These functions aren't used.

**Pull request merged**